src/mpid/ch4/shm/src/shm_init.c

```diff
@@ -23,6 +24,9 @@ int MPIDI_SHMI_mpi_init_hook(int rank, int size, int *n_vcis_provided, int *tag_
     if (ret)
         MPIR_ERR_POP(ret);
 #ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
+    ret = MPIR_SHM_coll_init(rank, size, MPIDI_global.node_map[0]);
+    if (ret)
+        MPIR_ERR_POP(ret);
 
```

Comment:
Why do we initialize the `SHM_coll` component in this commit ? If it is a general utility function for all SHM modules, the initialization/finalize should be in a separate commit. @gcongiu could you please confirm if my understanding is correct ?

Reply:
Discussed with @gcongiu offline. The `SHM_coll` utility will be changed to : (1) initialize a global shm segment at beginning of `MPI_Init`; (2) directly expose the address of global shm segment to the caller routine so that the caller can use it to exchange data without extra copy.

Reply:
@KaimingOuyang The new implementation of `shm data exchange` routine has been merged into master. Can you please check the commits and adjust the XPMEM code ?

Reply:
Why do we call `MPIDU_Init_shm_init ` only when XPMEM is enabled ? Should it be always initialized as it is used in multiple places at MPI init time ? Besides, is it correct to initialize it inside `MPIDI_SHMI_mpi_init_hook` ? Will it be used somewhere even when shm is disabled ? Same question for `MPIDU_Init_shm_finalize `. @gcongiu Could you please confirm the design for us ?

Reply:
It can also be used by business card exchange so it will be initialized also when XPMEM and SHM are not initialized